amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/persistence/mapper/TableMetaMapper.java

```diff
@@ -424,4 +425,97 @@ List<ServerTableIdentifier> selectTableIdentifiersByCatalog(
         typeHandler = MapLong2StringConverter.class)
   })
   List<TableRuntimeMeta> selectTableRuntimeMetas();
+
+  @Select(
```

Comment:
How about using case when in where to make these two interfaces as one.

If sorting is necessary, we could extend only one method

Reply:
will change using `case when`